### ISSUE TITLE
Hubble dynamic export doesn't respect compression settings

### ISSUE THREAD

**Issue by jcmcken** [cilium/cilium#36669]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Version

equal or higher than v1.15.11 and lower than v1.16.0

### What happened?

With ``hubble-export-file-compress`` enabled, and using the dynamic exporter, the rotated log files are not compressed. Things like max file backups and max log size do work. Looking over the [code](https://github.com/cilium/cilium/blob/v1.15.11/pkg/hubble/exporter/dynamic_exporter.go) I'm not sure I spot anywhere where the compression would conceivably be enabled for this exporter. Looking at more recent versions, I also don't notice where the compression config is referenced.

It's possible this would be considered a feature request rather than a bug. The docs do mention specifically that max files and max size settings work with dynamic exporter. But it wasn't necessarily obvious that compression or other settings wouldn't work. So maybe just needing a clearer admonition in the docs is the bug here.

### How can we reproduce the issue?

1. Install the Helm chart with Hubble enabled using something like:

```yaml
extraConfig:
  hubble-export-file-compress: "true"
hubble:
  export:
    fileMaxBackups: 1
    fileMaxSizeMb: 1
    dynamic:
      enabled: true
```

2. Generate network traffic to fill up the flow logs
3. Shell into one of the ``cilium`` pods and look at the log files in ``/var/run/cilium/hubble/``. The rotated log file (``events-2024-*.log``) is not compressed and is 1MB in size.

### Cilium Version

1.15.11

### Kernel Version

Linux 6.1.112

### Kubernetes Version

1.30.6

### Relevant log output

```shell

```

### Cilium Users Document

- [x] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by squeed** [comment]
Agreed, this would be a nice enhancement, but, as you say, it's not really a bug. The dynamic configuration is a completely different beast.

**Comment by chaunceyjiang** [comment]
Please assign to me.

I want to try to implement this feature.
